### DISCUSSION TITLE
Upload HTML files to hosting platform

### DISCUSSION THREAD

**Discussion by EdwardKS** [KosmosisDire/obsidian-webpage-export#351] [Q&A]
Hi Nathan,

Can you please send me the instruction on **How to upload the converted HTML files**?

I understand the first part … Convert markdown files to HTML files.  

Just want to make sure I follow the instructions on part 2 … upload the files to the website.

Thank you!

**Comment by KosmosisDire** [comment]
Hi, this is mostly left up to you, I do want to add some supported instructions for how to do this in the future. Right now I would recommend looking into vercel or github pages. There are tutorials online for how to set these up if you already have html files.

If you need help with a specific part you can certainly ask and I'll do my best to help.

**Comment by EdwardKS** [comment]
Hi Nathan,,

Method to upload:
Which method will be easier to deploy the html files?

New files:
When we upload new html files, do we need to re-upload the existing files again?

Edit files:
How do we edit the existing html files?  
Do we just re-upload the new version and let it overwrite the existing?

Thank you!

**Comment by KosmosisDire** [comment]
Method to upload:
I would reccomned using github pages or vercel. 
https://pages.github.com/
https://vercel.com/

I would reccomend setting it up so you don't have to manually upload your files each time you publish, this can be achieved using git. Both github pages and vercel support git. Just create a git repo, clone the repo, and drag your files into it. Then push your changes. It will keep track of which files need to be pushed to the repository for you.

New files:
When you create a new file or add a new file to the export that was not exported previously you should just export everything like normal. It will only export the new files and update all the metadata for the site. So in other words, just select all the files you would like on you site. Any time you make any change you want to publish just export it again. No need to change any settings or file selections, ecept to select any new files.

You technically only have to upload files which have changed, but due to the way the export is organized multiple files will change when you change a single file. So I would reccomnd using some kind of version control like git.

Edit files:
When you edit a file it will recreate only the files which have been edited. Again all you have to do is export again. Same as above.

**Comment by KosmosisDire** [comment]
(I'll also mention that if you'd like everything to work cleanly, you should probably use the beta version)

**Comment by ohmygaugh-crypto** [comment]
KosmosisDire, I realized my comment on an open issue is probably better documented in this thread here. Here is the link to it: [https://github.com/KosmosisDire/obsidian-webpage-export/issues/373#issuecomment-1962969819](https://github.com/KosmosisDire/obsidian-webpage-export/issues/373#issuecomment-1962969819 )

This is what I asked:

> 
> 
> can you please provide an example of the build folder for doing this manually?
> 
> In my folder I'm uploading to netlify drop I have:
> 
> "canvasfilename".html
> obsidian.css
> ?.js 

just trying to figure out which .js file and .css file you are using in conjunction with the .html export the plugin provides. 

I love how high quality this project is! Glad I stumbled across this repo. I was previously working on using a repo called: [BrainPress](https://github.com/zachdoty/BrainPress)

**Reply by KosmosisDire** [reply]
This fix unfortunately will not work for non web exports. It's possible to manually fix the self contained exports. You need to edit the .html files and add a style tag and paste the contents of an old working obsidian.css file.

I should have the actual fix out today
